tests/validation/cattlevalidationtest/core/test_machine.py

```diff
@@ -0,0 +1,264 @@
+from common_fixtures import *  # NOQA
+
+DEFAULT_TIMEOUT = 900
+
+# Digital Ocean configurations
+access_key = os.environ.get('DIGITALOCEAN_KEY')
+image_name = "ubuntu-14-10-x64"
+region = "sfo1"
+size = "2gb"
+
+# Digital Ocean default configurations
+default_size = "512mb"
+default_image_name = "ubuntu-14-04-x64"
+default_region = "nyc3"
+
+
+# Digital Ocean Error Messages
+error_msg_auth_failure = "Error creating machine: " \
+                         "GET https://api.digitalocean.com/v2/regions:" \
+                         " 401 Unable to authenticate you."
+
+error_msg_invalid_region = "Error creating machine: " \
+                           "digitalocean requires a valid region"
+
+
+if_machine_digocean = pytest.mark.skipif(
+    os.environ.get('DIGITALOCEAN_KEY') is None,
+    reason='DIGITALOCEAN_KEY is not set')
+
+
+@if_machine_digocean
+def test_digital_ocean_machine_all_params(client):
+
+    name = random_str()
+    create_args = {"name": name,
+                   "digitaloceanConfig": {"accessToken": access_key,
+                                          "image": image_name,
+                                          "region": region,
+                                          "size": size
+                                          }
+                   }
+
+    expected_values = {"image": image_name,
+                       "region": region,
+                       "size": size
+                       }
+    try:
+        digital_ocean_machine_life_cycle(client, create_args, expected_values)
+    finally:
+        delete_host_in_digital_ocean(name)
+
+
+@if_machine_digocean
+def test_digital_ocean_machine_accesstoken(client):
```

Comment:
There may be some value in making sure that the default values are checked since we have them set as defaults in our UI . If defaults change in different versions of docker-machines , we will be aware of these changes when the test validation breaks. This will help us fix defualts in UI as well.